pkg/internal/tests/pulcheck/pulcheck.go

```diff
@@ -75,7 +89,7 @@ func EnsureProviderValid(t T, tfp *schema.Provider) {
 			}
 		}
 
-		if resourceNeedsUpdate(r) && r.UpdateContext == nil {
+		if r.UpdateContext == nil {
```

Comment:
Why do we need to add the `if !resourceNeedsUpdate(r) { ... }` block if we remove the conditional.